pkg/cmd/config/config.go

```diff
@@ -85,6 +97,18 @@ func NewCmdConfigSet(f *cmdutil.Factory) *cobra.Command {
 			}
 
 			key, value := args[0], args[1]
+			knownKey := false
+			for _, configKey := range respectedConfigKeys {
+				if key == configKey.name {
+					knownKey = true
+					break
+				}
+			}
+			if !knownKey {
+				iostreams := f.IOStreams
+				warningIcon := iostreams.ColorScheme().WarningIcon()
+				fmt.Fprintf(iostreams.ErrOut, "%s warning: '%s' is not a known configuration key\n", warningIcon, key)
+			}
```

Comment:
What are your thoughts on moving this code into `cfg.Set`? We have another PR that is doing config value validation (https://github.com/cli/cli/pull/2138) and I think it would be nice to have both the key and value validations residing in the same place.

Reply:
I'm actually fine with this being here and not within `Set()` since it's only a warning. Other potential callers of the `Set()` function should be able to set arbitrary keys without it resulting in a warning printed on stderr.

Although, I do agree with @samcoe that this check is similar to config value validation and could thus live in the `config` package. Perhaps for now, the logic that searches for a key being among `respectedConfigKeys` could be moved to the `config` package and invoked from here like `config.IsSupportedConfig(key)`.

Reply:
Just wanted to update here that I merged in #2138 👍